src/transformers/trainer.py

```diff
@@ -4291,11 +4296,10 @@ def evaluation_loop(
 
         # if full fp16 or bf16 eval is wanted and this ``evaluation`` or ``predict`` isn't called
         # while ``train`` is running, cast it to the right dtype first and then put on device
-        if not self.is_in_train:
-            if args.fp16_full_eval:
-                model = model.to(dtype=torch.float16, device=args.device)
-            elif args.bf16_full_eval:
-                model = model.to(dtype=torch.bfloat16, device=args.device)
+        if not self.is_in_train and args.mixed_precision_config["full_eval"]:
+            model = model.to(
+                dtype=torch.float16 if args.mixed_precision_dtype == "fp16" else torch.bfloat16, device=args.device
```

Comment:
Here the logic is slightly changed vs before, where the model was only cast to `bfloat16` if the dtype was explicitly `bfloat16`, whereas now `bfloat16` is just the `else` option.